runtime/compiler/codegen/J9TreeEvaluator.cpp

```diff
@@ -595,7 +595,7 @@ uint32_t getInstanceOfOrCheckCastTopProfiledClass(TR::CodeGenerator *cg, TR::Nod
    TR_ByteCodeInfo bcInfo = node->getByteCodeInfo();
    TR_ValueProfileInfoManager *valueProfileInfo = TR_ValueProfileInfoManager::get(comp);
 
-   if (!valueProfileInfo)
+   if (!valueProfileInfo || comp->compileRelocatableCode())
```

Comment:
I think this addition deserves a comment. _Why_ is this disabled for AOT? The commit message answers the question but it may get buried over time. I think future readers will appreciate a comment here.

Reply:
Added comment in https://github.com/eclipse/openj9/compare/32cebb26bab96e1e2dae7023f04be4ea5ac833ee..2299020f96bf446ca21f325cc3ff7c8e3f412281